adr/2025_03_10-suggestions/suggestions-as-history.md

```diff
@@ -0,0 +1,31 @@
+# Storing changes as separate _document histories_ and diffs between them
+
+Branched document histories is the idea that you can think of a suggestion of the document as having a "different history" of a document; where, a user made some changes to a copy of the document, which can later be merged back into the document somehow.
+
+![Branched document histories](./images/doc+suggestions.png)
+
+This would work very similarly to how the version diffing would work, and is intended to mostly re-use that logic.
+
+## Implementation
+
+To implement this, each user who suggests changes is given a separate document which they would have access to modify the contents of. This allows the user to own that document, while guaranteeing the original document stays unmodified.
+
+![Branched document histories merged](./images/doc+suggestions_merged.png)
+
+When viewing what suggestions are available on a document, the frontend would have to "collect" the suggestions from each suggestion document, apply an updates those documents may not have seen yet, diff the documents to see what changes are in the suggestion document compared to the original document and finally display those changes to the user in the original document somehow.
```

Comment:
Kevin suggested that we instead have a server-maintained "suggestions doc" where all suggestions would be recorded into. Rather than having to have something which collects these suggestions into one document.